### PR TITLE
new styleguide

### PR DESCRIPTION
**Pull request by tarikEast**
Pull request for issue https://github.com/itsinthenode/styleguide/issues/3

### CODE REVIEW

File: assets/stylesheets/forms/_buttons.scss [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+// Buttons
+//
+// .primary-lime - Primary lime button
+// .primary-royal-blue - Primary royal blue button
+// .secondary-metal - Secondary metal button
+// .secondary-abort - Secondary abort button
+//
+// Markup:
+// <button class="itn-btn {{modifier_class}}">Text only</button>
```
Comment (deepflame) [inline assets/stylesheets/forms/_buttons.scss]: internally we already use `nd` as a "itsintheNoDe" prefix. Could you change this here as well?

File: assets/stylesheets/forms/_buttons.scss [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+//   <use xlink:href="#set1-comment" />
+//  </svg> Link
+// </a>
+//
+// Weight: 2
+//
+// Style guide: Buttons.Links
+
+a {
+    line-height: 28px;
```
Comment (deepflame) [inline assets/stylesheets/forms/_buttons.scss]: line-height and font-size could/should come from a text style, no?

File: assets/stylesheets/forms/_inputs.scss [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+//
+// Style guide: FormElements.Inputs
+
+$border-color: #29384A;
+$border-color-shadow: black;
+$placeholder-color: #888FA8;
+$focus-color: #2278F1;
+$focus-color-shadow: #CBE6FF;
+
+@mixin inputStyle{
```
Comment (deepflame) [inline assets/stylesheets/forms/_inputs.scss]: please use `dashes` for naming mixins and everyting CSS related. Javascript has `camelcase` as here.

File: assets/stylesheets/forms/_inputs.scss [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    font-size: 14px;
+    height: 40px;
+    color: color('carbon');
+    padding: 0 15px; 
+    color: color(carbon);  
+    margin-bottom: 10px; 
+    line-height: 100%;
+    outline: none;
+
+    ::-webkit-input-placeholder{
```
Comment (deepflame) [inline assets/stylesheets/forms/_inputs.scss]: I would think that this is not necessary in out desktop and mobile version as we use https://github.com/kimroen/ember-cli-autoprefixer there. Maybe we should implement it into the styleguide here as well?

(but not really necessary as we use it in the Rails project also, just an idea to make it consistent)

File: assets/stylesheets/forms/_inputs.scss [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+        }
+    }
+
+    select {
+        display: block;
+        width: 100%;
+        height: 40px;
+        background-color: white;
+        appearance: none;
+        -moz-appearance: none;
```
Comment (deepflame) [inline assets/stylesheets/forms/_inputs.scss]: please do not use vendor prefixes as we use autoprefixer.

**Comment by deepflame** [comment]
@tarikEast I added an `editorconfig` file. Please use these settings for your editor, thanks. (http://editorconfig.org/)

File: custom-template/package.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (tarikEast) [inline custom-template/package.json]: I have added the custom template so I can inline svg definitions into head of index. I will try to inline svg definitions somehow different, maybe to have section icon definitions that has markup of svg icons. Will look into it.

File: assets/images/icons/input/select-arrow.svg [hunk 7]
```diff
@@ -0,0 +1,38 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="20" height="17" xmlns:xlink="http://www.w3.org/1999/xlink">
+  <desc>
+    Created using Figma
```
Comment (tarikEast) [inline assets/images/icons/input/select-arrow.svg]: I think I did. Will try to clean it up manually.

File: assets/stylesheets/forms/_inputs.scss (hunk 4 above)
Comment (tarikEast) [inline assets/stylesheets/forms/_inputs.scss]: will remove prefixes for vendors, no problem  :+1:

File: custom-template/package.json (hunk 6 above)
Comment (tarikEast) [inline custom-template/package.json]: found the other way to inline svgs (through Markup template of node-kss), will remove this

**Comment by deepflame** [comment]
TODO: check changes from Ivo and e.g. put colors into variables
https://github.com/itsinthenode/styleguide/pull/4/commits/5d85beaea3bd2c1b042bb9ef0c4e44b9b407c5bd?w=1

**Comment by deepflame** [comment]
since we have some issues with the new styleguide we will merge this into 'development' now and continue work from there.

**Pull request merged**